src/packageurl/contrib/url2purl.py

```diff
@@ -276,3 +294,100 @@ def build_sourceforge_purl(uri):
 @purl_router.route(cargo_pattern)
 def build_cargo_purl(uri):
     return purl_from_pattern('cargo', cargo_pattern, uri)
+
+
+@purl_router.route('https?://github.com/.*')
+def build_github_purl(uri):
```

Comment:
Repasting here for context:

I agree with @majurg .. we want to accept inputs in a flexible way. In fact this is the whole purpose there: take possibly messy URLs and turn them in a clean purl.
also there could be many other cases beyond www prefix?
What about `https://raw.githubusercontent.com/volatilityfoundation/dwarf2json/master/LICENSE.txt` or API URLs or  `https://codeload.github.com/nexB/scancode.io/tar.gz/1.0` 

So for clarity we should have multiple patterns here.

 Also even the current form may not be correct. Why `.*` right after `.com`? and the dot in `.com`?
This looks better imho (but we still ned to accommodate more  as stated above)
`@purl_router.route('https?://github\.com/.*')`